source/Octopus.Tentacle/Kubernetes/IKubernetesPodLogService.cs

```diff
@@ -107,5 +185,36 @@ public KubernetesPodLogService(IKubernetesClientConfigProvider configProvider, I
                 }
             }
         }
+
+        class WrappedProcessOutput : ProcessOutput
+        {
+            static readonly TimeSpan OneTick = TimeSpan.FromTicks(1);
+            readonly string wrapper;
+
+            public WrappedProcessOutput(ProcessOutputSource source, string text, DateTimeOffset occurred, string wrapper)
+                : base(source, text, occurred)
+            {
+                this.wrapper = wrapper;
+            }
+
+            public IEnumerable<ProcessOutput> Expand()
+            {
+                return new[]
+                {
+                    //we add the service messages one tick before and after so they are correctly ordered
+                    new ProcessOutput(ProcessOutputSource.StdOut, $"##octopus[stdout-{wrapper}]", Occurred.Subtract(OneTick)),
+                    new ProcessOutput(Source, Text, Occurred),
```

Comment:
If there are events that occur in the same microsecond, could this cause issues with service message formatting?

Reply:
yeah, technically... I couldn't think of a better way to do this... maybe in the same second... I'm just concerned if _some other code_ does a sort based on `Occurred`